scene/3d/ray_cast_3d.cpp

```diff
@@ -318,6 +342,77 @@ void RayCast3D::_bind_methods() {
 	ADD_GROUP("Collide With", "collide_with");
 	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "collide_with_areas", PROPERTY_HINT_LAYERS_3D_PHYSICS), "set_collide_with_areas", "is_collide_with_areas_enabled");
 	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "collide_with_bodies", PROPERTY_HINT_LAYERS_3D_PHYSICS), "set_collide_with_bodies", "is_collide_with_bodies_enabled");
+
+	ADD_GROUP("Debug Shape", "debug_shape");
+	ADD_PROPERTY(PropertyInfo(Variant::COLOR, "debug_shape_color"), "set_debug_shape_color", "get_debug_shape_color");
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "debug_shape_thickness", PROPERTY_HINT_RANGE, "1,20"), "set_debug_shape_thickness", "get_debug_shape_thickness");
+}
+
+float RayCast3D::get_debug_shape_thickness() const {
+	return debug_shape_thickness;
+}
+
+void RayCast3D::_update_debug_shape_vertex() {
+	if (target_position == Vector3()) {
+		debug_shape_vertex.clear();
+		return;
+	}
+
+	debug_shape_vertex.clear();
+
+	if (debug_shape_thickness <= 1) {
+		//Draw line
+		debug_shape_vertex.push_back(Vector3());
+		debug_shape_vertex.push_back(target_position);
+	} else {
+		float scale_factor = 100.0;
+		Vector3 dir = Vector3(target_position).normalized();
+		// Draw parallelepiped
+		Vector3 normal = (fabs(dir.x) + fabs(dir.y) > CMP_EPSILON) ? Vector3(-dir.y, dir.x, 0).normalized() : Vector3(0, -dir.z, dir.y).normalized();
+		normal *= debug_shape_thickness / scale_factor;
+		int vertices_strip_order[14] = { 4, 5, 0, 1, 2, 5, 6, 4, 7, 0, 3, 2, 7, 6 };
+		for (int v = 0; v < 14; v++) {
+			Vector3 vertex = vertices_strip_order[v] < 4 ? normal : normal / 3.0 + target_position;
+			debug_shape_vertex.push_back(vertex.rotated(dir, Math_PI * (0.5 * (vertices_strip_order[v] % 4) + 0.25)));
+		}
+	}
+}
+
+void RayCast3D::set_debug_shape_thickness(const float p_debug_shape_thickness) {
+	debug_shape_thickness = p_debug_shape_thickness;
+	if (debug_shape && !Engine::get_singleton()->is_editor_hint()) {
+		_clear_debug_shape();
+		_update_debug_shape();
+	}
+	if (is_inside_tree() && (Engine::get_singleton()->is_editor_hint())) {
+		_update_debug_shape_vertex();
+		update_gizmo();
+	}
+}
+
+Vector<Vector3> RayCast3D::get_debug_shape_vertex() {
+	return debug_shape_vertex;
+}
+
+void RayCast3D::set_debug_shape_color(const Color p_color) {
+	debug_shape_color = p_color;
+	if (debug_material.is_valid()) {
+		Ref<StandardMaterial3D> line_material = static_cast<Ref<StandardMaterial3D>>(debug_material);
+		line_material->set_shading_mode(StandardMaterial3D::SHADING_MODE_UNSHADED);
+		line_material->set_albedo(debug_shape_color);
+		line_material->set_transparency(BaseMaterial3D::TRANSPARENCY_ALPHA);
+	}
+}
+
+Ref<StandardMaterial3D> RayCast3D::get_debug_material() {
+	if (!debug_material.is_valid()) {
```

Comment:
Here based on my comment with `_update_debug_shape_material`, you could just do:
```
Ref<StandardMaterial3D> RayCast3D::get_debug_material() {
	_update_debug_shape_material();
	return debug_material;
}
```